packages/jira-adapter/src/reference_mapping.ts

```diff
@@ -959,6 +959,48 @@ export const referencesRules: JiraFieldReferenceDefinition[] = [
     jiraMissingRefStrategy: 'typeAndValue',
     target: { type: PROJECT_TYPE },
   },
+  {
+    src: { field: 'projectKey', parentTypes: [CUSTOMER_PERMISSIONS_TYPE] },
+    jiraSerializationStrategy: 'key',
+    jiraMissingRefStrategy: 'typeAndValue',
+    target: { type: PROJECT_TYPE },
+  },
+  {
+    src: { field: 'projectKey', parentTypes: [QUEUE_TYPE] },
+    jiraSerializationStrategy: 'key',
+    jiraMissingRefStrategy: 'typeAndValue',
+    target: { type: PROJECT_TYPE },
+  },
+  {
+    src: { field: 'projectKey', parentTypes: [REQUEST_TYPE_NAME] },
+    jiraSerializationStrategy: 'key',
+    jiraMissingRefStrategy: 'typeAndValue',
+    target: { type: PROJECT_TYPE },
+  },
```

Comment:
pretty sure you can combine these (plus the two below)
```suggestion
  {
    src: { field: 'projectKey', parentTypes: [CUSTOMER_PERMISSIONS_TYPE, QUEUE_TYPE, REQUEST_TYPE_NAME, REQUEST_TYPE_GROUP_TYPE, CALENDAR_TYPE] },
    jiraSerializationStrategy: 'key',
    jiraMissingRefStrategy: 'typeAndValue',
    target: { type: PROJECT_TYPE },
  },
```